packages/react-storage/src/components/StorageBrowser/views/LocationActionView/hooks/useDeleteActionView.tsx

```diff
@@ -0,0 +1,219 @@
+import { humanFileSize, isFunction } from '@aws-amplify/ui';
+
+import { deleteHandler } from '../../../actions/handlers';
+import { TaskCounts } from '../../../controls/types';
+import {
+  DataTableProps,
+  DataTableRow,
+} from '../../../composables/DataTable/DataTable';
+import { WithKey } from '../../../components/types';
+import { getTaskCounts } from '../../../controls/getTaskCounts';
+import { STATUS_DISPLAY_VALUES } from '../../LocationActionView/constants';
+import { useStore } from '../../../providers/store';
+import { useGetActionInput } from '../../../providers/configuration';
+import { Task, useProcessTasks } from '../../../tasks';
+import { UseActionView } from './types';
+import {
+  getActionIconVariant,
+  getActionViewTaskStatuses,
+  getFilenameWithoutPrefix,
+  getFileTypeDisplayValue,
+} from './utils';
+
+interface UseDeleteActionView extends UseActionView {}
+
+const LOCATION_ACTION_VIEW_HEADERS: DataTableProps['headers'] = [
+  { key: 'key', type: 'sort', content: { label: 'Name' } },
+  { key: 'folder', type: 'text', content: { text: 'Folder' } },
+  { key: 'type', type: 'text', content: { text: 'Type' } },
+  { key: 'size', type: 'text', content: { text: 'Size' } },
+  { key: 'status', type: 'sort', content: { label: 'Status' } },
+  { key: 'action', type: 'text', content: { text: '' } },
+];
+
+export const getDeleteActionViewTableData = ({
+  tasks,
+  taskCounts,
+  path,
+}: {
+  tasks: Task[];
+  taskCounts: TaskCounts;
+  path: string;
+}): DataTableProps => {
+  const { hasStarted } = getActionViewTaskStatuses(taskCounts);
+
+  const rows: DataTableProps['rows'] = tasks.map((item) => {
+    const row: WithKey<DataTableRow> = {
+      key: item.key,
+      content: LOCATION_ACTION_VIEW_HEADERS.map(({ key: columnKey }) => {
+        const key = `${columnKey}-${item.key}`;
+        switch (columnKey) {
```

Comment:
Can we move this switch statement to a util? Having it here makes this hook v hard to read over

Reply:
done